lib/include/irritator/core.hpp

```diff
@@ -5220,14 +5220,76 @@ struct constant
     }
 };
 
-struct filter
-{
+struct filter {
     port x[1];
     port y[1];
-    time sigma = time_domain<time>::infinity;
-    double current_value = 0.0;
-    double up_threshold = 0.0;
-    double down_threshold = 0.0;
+    time sigma;
+
+    double default_lower_threshold;
+    double default_upper_threshold;
+
+    double lower_threshold;
+    double upper_threshold;
+
+    double default_value;
+    double default_input_coeff;
+
+    double value;
+    double input_coeff;
+
+    //filter() noexcept { 
+    //    default_lower_threshold=0.0;
+    //    default_upper_threshold = 1.0;
+    //    default_input_coeff=1.0;
+    //    default_value=0.0;
+
+    //}
+
+    status initialize() noexcept { 
+        default_lower_threshold = 0.0;
+        default_upper_threshold = 1.0;
+        default_input_coeff = 1.0;
+        default_value = 0.0;
+        lower_threshold=default_lower_threshold;
+        upper_threshold=default_upper_threshold;
+        input_coeff=default_input_coeff;
```

Comment:
```
        default_input_coeff = 1.0;
        input_coeff=default_input_coeff;
```

later
`            to_send = value*input_coeff;`

May be `default_input_coeff` and `input_coeff` are useless no ?